src/Tokenizer/Analyzer/Analysis/ArgumentAnalysis.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+/**
+ * @internal
+ */
+final class ArgumentAnalysis
+{
+    /**
+     * The default value of the argument.
+     *
+     * @var null|string
+     */
+    private $default;
+
+    /**
+     * The name of the argument.
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * The index where the name is located in the supplied Tokens object.
+     *
+     * @var int
+     */
+    private $nameIndex;
+
+    /**
+     * The type of the argument.
+     *
+     * @var null|string
+     */
+    private $type;
```

Comment:
Follow up to my scalar type coment above. Would it be worth it to add `isScalar()` method here?

I'm working with types on many places and I have to check this on every space manually. This method would save repeated code.